### ISSUE TITLE
‘‘libsoft_oal’’ is missing.but I have openal

### ISSUE THREAD

**Issue by kero990** [OpenRA/OpenRA#19750]
I compiled Openra on an arm64-based PC and installed all dependencies according to the instructions. The compilation was successful, but it crashed during runtime. The log file is attached.
It seems that it is an openal problem, and it seems that a dependency called ‘‘libsoft_oal’’ is missing. But in the list of files I downloaded from the openal official website and compiled and installed, there is no library file starting with libsoft.
What should I do?

```
System.DllNotFoundException: Unable to load shared library 'soft_oal' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: libsoft_oal: cannot open shared object file: No such file or directory
   at OpenAL.ALC10.alcOpenDevice(String devicename)
   at OpenRA.Platforms.Default.OpenAlSoundEngine..ctor(String deviceName) in /home/inspur/OpenRA/OpenRA.Platforms.Default/OpenAlSoundEngine.cs:line 124
   at OpenRA.Platforms.Default.DefaultPlatform.CreateSound(String device) in /home/inspur/OpenRA/OpenRA.Platforms.Default/DefaultPlatform.cs:line 28
   at OpenRA.Sound..ctor(IPlatform platform, SoundSettings soundSettings) in /home/inspur/OpenRA/OpenRA.Game/Sound/Sound.cs:line 52
   at OpenRA.Game.Initialize(Arguments args) in /home/inspur/OpenRA/OpenRA.Game/Game.cs:line 369
System.DllNotFoundException: Unable to load shared library 'soft_oal' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: libsoft_oal: cannot open shared object file: No such file or directory
   at OpenAL.ALC10.alcOpenDevice(String devicename)
   at OpenRA.Platforms.Default.OpenAlSoundEngine..ctor(String deviceName) in /home/inspur/OpenRA/OpenRA.Platforms.Default/OpenAlSoundEngine.cs:line 124
   at OpenRA.Platforms.Default.DefaultPlatform.CreateSound(String device) in /home/inspur/OpenRA/OpenRA.Platforms.Default/DefaultPlatform.cs:line 28
   at OpenRA.Sound..ctor(IPlatform platform, SoundSettings soundSettings) in /home/inspur/OpenRA/OpenRA.Game/Sound/Sound.cs:line 52
   at OpenRA.Game.Initialize(Arguments args) in /home/inspur/OpenRA/OpenRA.Game/Game.cs:line 369

```

**Comment by matjaeck** [comment]
I have found something similar on https://magazine.odroid.com/article/playing-modern-fna-games-on-the-odroid-platform/, where it specifically mentions ARM and says:
> One thing to note is that some dll’s want to load “oal_soft.dll”. It should be redirected to “libopenal.so” but somehow, it is not. Easier way is to create a symlink inside Stardew valley folder to “libopenal.so” named “liboal_soft.so” and it will work. On my N1, which is ARM64, the command would be:
$ cd ~/StardewValley
$ ln -s /usr/lib/aarch64-linux-gnu/libopenal.so libsoft_oal.so

I don't know if that's any help.

**Comment by pchote** [comment]
Can you please paste the output of running `./configure-system-libraries.sh` in OpenRA's root dir?

**Comment by kero990** [comment]
According to the above instructions, I solved two dependency problems with different names. Now I can install and enter the game. But the game will crash the moment I select MCV. Attach the log again, thank you very much.
In addition, no matter before or now, no matter whether I correctly'export' the .NET environment in the terminal, there is no response from'./configure-system-libraries.sh'.
Also, after running ./launch-game.sh, the terminal always reported an error: (zenity:12380): GLib-GObject-CRITICAL **: 15:02:06.099: g_value_set_boxed: assertion'G_VALUE_HOLDS_BOXED (value)' failed. I think my environment still has problems.

Attached the log of the crash：

```
OpenRA engine version {DEV_VERSION}
Red Alert mod version {DEV_VERSION}
on map bc4f526b46e420a1c5f7d645eb129a2fc2b0cae2 (Ridges by morkel).
Date: 2021-10-18 06:56:09Z
Operating System: Linux (Unix 4.19.0.64)
Runtime Version: .NET CLR 5.0.11
Exception of type `System.InvalidOperationException`: The current order generator may not be changed from synced code
   at OpenRA.Sync.AssertUnsynced(String message) in /home/inspur/OpenRA/OpenRA.Game/Sync.cs:line 208
   at OpenRA.World.set_OrderGenerator(IOrderGenerator value) in /home/inspur/OpenRA/OpenRA.Game/World.cs:line 153
   at OpenRA.World.CancelInputMode() in /home/inspur/OpenRA/OpenRA.Game/World.cs:line 162
   at OpenRA.Mods.Common.Widgets.WorldInteractionControllerWidget.HandleMouseInput(MouseInput mi) in /home/inspur/OpenRA/OpenRA.Mods.Common/Widgets/WorldInteractionControllerWidget.cs:line 163
   at OpenRA.Widgets.Widget.HandleMouseInputOuter(MouseInput mi) in /home/inspur/OpenRA/OpenRA.Game/Widgets/Widget.cs:line 424
   at OpenRA.Widgets.Ui.HandleInput(MouseInput mi) in /home/inspur/OpenRA/OpenRA.Game/Widgets/Widget.cs:line 100
   at OpenRA.DefaultInputHandler.<>c__DisplayClass5_0.<OnMouseInput>b__0() in /home/inspur/OpenRA/OpenRA.Game/Input/InputHandler.cs:line 50
   at OpenRA.DefaultInputHandler.OnMouseInput(MouseInput input) in /home/inspur/OpenRA/OpenRA.Game/Input/InputHandler.cs:line 51
   at OpenRA.Platforms.Default.Sdl2Input.PumpInput(Sdl2PlatformWindow device, IInputHandler inputHandler, Nullable`1 lockedMousePosition) in /home/inspur/OpenRA/OpenRA.Platforms.Default/Sdl2Input.cs:line 149
   at OpenRA.Platforms.Default.Sdl2PlatformWindow.PumpInput(IInputHandler inputHandler) in /home/inspur/OpenRA/OpenRA.Platforms.Default/Sdl2PlatformWindow.cs:line 459
   at OpenRA.Renderer.EndFrame(IInputHandler inputHandler) in /home/inspur/OpenRA/OpenRA.Game/Renderer.cs:line 312
   at OpenRA.Game.RenderTick() in /home/inspur/OpenRA/OpenRA.Game/Game.cs:line 728
   at OpenRA.Game.Loop() in /home/inspur/OpenRA/OpenRA.Game/Game.cs:line 832
   at OpenRA.Game.Run() in /home/inspur/OpenRA/OpenRA.Game/Game.cs:line 861
   at OpenRA.Game.InitializeAndRun(String[] args) in /home/inspur/OpenRA/OpenRA.Game/Game.cs:line 294
   at OpenRA.Launcher.Program.Main(String[] args) in /home/inspur/OpenRA/OpenRA.Launcher/Program.cs:line 32

```

**Comment by abcdefg30** [comment]
That crash is already fixed for latest bleed in #19741.

**Comment by abcdefg30** [comment]
Closing as resolved.
